esmvalcore/preprocessor/__init__.py

```diff
@@ -277,29 +277,64 @@ def _get_multi_model_settings(products, step):
     return settings, exclude
 
 
-def _run_preproc_function(function, items, kwargs):
+def _run_preproc_function(function, items, kwargs, input_files=None):
     """Run preprocessor function."""
-    msg = "{}({}, {})".format(function.__name__, items, kwargs)
-    logger.debug("Running %s", msg)
+    if input_files is None:
+        file_msg = ""
+    else:
+        file_msg = f"\nOriginal input file(s):\n{pformat(input_files)}"
+    logger.debug("Running %s with option(s)\n%s\non argument(s)\n%s%s",
+                 function.__name__, pformat(kwargs), pformat(items), file_msg)
     try:
         return function(items, **kwargs)
     except Exception:
+        # To avoid very long error messages, we truncate the arguments and
+        # input files here at a given threshold
+        n_shown_args = 4
+        if input_files is None:
+            file_msg = ""
+        elif len(input_files) <= n_shown_args:
+            file_msg = f"\nOriginal input file(s):\n{pformat(input_files)}"
+        else:
```

Comment:
```suggestion
        if len(input_files) > n_shown_args:
```
the other two cases duplicate the code from lines 282 to 285